### ISSUE TITLE
SC Pairing fails on macOS Catalina 

### ISSUE THREAD

**Issue by drazenzubovic** [OpenSC/OpenSC#1844]
### Problem Description

This issue started appearing after the fix of the driver in [OpenSC-0.20.0-rc3](https://github.com/OpenSC/OpenSC/releases/tag/OpenSC-0.20.0-rc3), which was a fix for issues reported under https://github.com/OpenSC/OpenSC/issues/1798

The above fix made certificates from the card visible and functional for macOS again. All except pairing functionality so far. After the card is inserted, the message appears from SmartCard Pairing: "Do you want to connect the inserted SmartCard with the current user?", with details of the Card identity, Reader (Gemalto PC Twin Reader) and the Driver (OpenSC token driver). Once the button "Pair" is pressed, and corresponding passwords entered, the message prompts that:

SmartCard pairing failed.
Cannot pair SmartCard with the current user.
The operation couldn’t be completed. (TKAuthenticationHintsProvider error -13 - The SmartCard certificate is not trusted.)

The log file below indicates that the certificate chain cannot be established. Is there any way to verify certificate chain and/or to trace/debug it? 

### Steps to reproduce

macOS: Catalina 10.15.1 (19B88)
OpenSC: OpenSC-0.20.0-rc3

### Logs

Activation of OpenSC logging does not show any trace of an error, presumably as the pairing functionality is part of the OS. The captured console output shows this:
```
ctkahp, Trust evaluate failure: [leaf MissingIntermediate]
ctkahp, Certificate could not be verified: Error Domain=NSOSStatusErrorDomain Code=-25318 "errKCCreateChainFailed / errSecCreateChainFailed:  / The attempt to create a certificate chain failed." UserInfo={NSLocalizedDescription=“MY-PKI-CERT” certificate is not trusted, NSUnderlyingError=0x7fe3dbc09af0 {Error Domain=NSOSStatusErrorDomain Code=-25318 "errKCCreateChainFailed / errSecCreateChainFailed:  / The attempt to create a certificate chain failed." UserInfo={NSLocalizedDescription=Certificate 0 “MY-PKI-CERT” has errors: Unable to build chain to root (possible missing intermediate);}}}
ctkahp, DTL error occured: Error Domain=TKAuthenticationHintsProvider Code=-13 "The SmartCard certificate is not trusted." UserInfo={NSLocalizedDescription=The SmartCard certificate is not trusted.}
ctkbind, pairUser = -5, error: Error Domain=TKAuthenticationHintsProvider Code=-13 UserInfo={NSDescription=<private>} (user = <private>, pubkey hash: <private>)
```

**Comment by frankmorgner** [comment]
OpenSC's debug options can be used if you [choose the correct location for the log file](https://github.com/frankmorgner/OpenSCToken/#configuring-opensctoken).

**Comment by frankmorgner** [comment]
Has this been working with OpenSCToken (*not* OpenSC.tokend) before https://github.com/frankmorgner/OpenSCToken/pull/15?

**Comment by frankmorgner** [comment]
By the way, the error message sounds like as if the trust in your smart card certificates is the problem, not the driver itself. Maybe you need to import your CA's certificate as trusted...

**Comment by mouse07410** [comment]
Must add that in the older macOS versions one had to add the trusted CAs via CLI - somehow adding them via Keychain Access did not have the desirable effect. Possibly, it's still true for Catalina.

**Comment by drazenzubovic** [comment]
@frankmorgner , @mouse07410 : It looks like the problem is entirely outside of OpenSC/OpenSCToken. 

When i try to verify the certificate chain from CLI (of the smart card certificate) i get the same error: Unable to find next certificate in the chain [MissingIntermediate]

The part i do not understand any more is, how could it work prior to Catalina? Smart card and certificates were the same?!

**Comment by frankmorgner** [comment]
Please use Apple's support to answer that question.

What would be the appropriate command to add the card's certificate to the trust store?

**Comment by mouse07410** [comment]
Apple has been tightening up certificate validation rules. I’ve told that in Catalina no certificate issued after July 2019 would be accepted if it’s life time is set for more than one year. I personally find such a policy non-smart, but nobody asked my opinion. So, it may be that one or more certs were OK by Mojave’s standards, but aren’t any more by Catalina’s. 

But, based on the error you’re getting - are all the certs in the chain displayed as valid/verified? It appears as if at least one in the chain is either absent, or fails validation.

**Comment by drazenzubovic** [comment]
Both hints have shown to be very valuable in this case! 

1) Check of the certificate chain was pointing to the missing intermediate. The problem got solved immediately once the intermediate was successfully imported into Keychain. 
2) Even on the latest macOS releases is Keychain tool not able to import trusted CA from the GUI reliably. I had to use CLI for this to happen indeed. 

Thank you very much for your support!
